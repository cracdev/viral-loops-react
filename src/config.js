const config = {
  viralLoopsKey: process.env.REACT_APP_VIRAL_LOOPS_KEY,
}

export default config;