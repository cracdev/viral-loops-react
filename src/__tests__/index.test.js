import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ViralLoops from '../index';
import config from '../config';

const setup = (propOverrides) => {
  const props = Object.assign({
    publicToken: ''
  }, propOverrides);

  const component = shallow(<ViralLoops {...props} />);

  return {
    component,
    props,
    formWidget: component.find({ 'data-test': 'ViralLoops_formWidget' }),
    milestoneWidget: component.find({ 'data-test': 'ViralLoops_milestoneWidget' }),
    counterWidget: component.find({ 'data-test': 'ViralLoops_counterWidget' }),
  }
}

describe('<ViralLoops />', () => {
  // The Viral Loops script requires a page with an existing <script> tag to
  // run without errors.
  document.body.innerHTML = 
    '<script>' +
    '</script>';
  require('./script');

  it('attaches the VL object to window', () => {
    const props = { 
      publicToken: config.viralLoopsKey,
    };
    setup(props);

    const viralLoopsObject = {
      instances: expect.any(Object),
      invoked: expect.any(Boolean),
      load: expect.any(Function),
    }

    expect(window.VL).toMatchObject(viralLoopsObject);
  });

  describe('formWidget', () => {
    const props = {
      publicToken: config.viralLoopsKey,
      formWidget: true,
    }
    const { component } = setup(props);

    it('renders with a form section', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('milestoneWidget', () => {
    const props = {
      publicToken: config.viralLoopsKey,
      milestoneWidget: true,
    }
    const { component } = setup(props);

    it('renders with a rewards section', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('counterWidget', () => {
    const props = {
      publicToken: config.viralLoopsKey,
      counterWidget: true,
    }
    const { component } = setup(props);

    it('renders with a counter section', () => {
      expect(component).toMatchSnapshot();
    });
  });
});