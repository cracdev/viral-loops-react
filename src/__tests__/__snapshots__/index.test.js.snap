// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViralLoops /> counterWidget renders with a counter section 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViralLoops
    counterWidget={true}
    publicToken="HaOj3wBPfSJhoWkoW2Ad_rmvPoM"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED>
        
        
        <div
          data-test="ViralLoops_counterWidget"
          data-vl-widget="referralCountWidget"
        />
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "",
          "",
          <div
            data-test="ViralLoops_counterWidget"
            data-vl-widget="referralCountWidget"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data-test": "ViralLoops_counterWidget",
            "data-vl-widget": "referralCountWidget",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED>
          
          
          <div
            data-test="ViralLoops_counterWidget"
            data-vl-widget="referralCountWidget"
          />
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "",
            "",
            <div
              data-test="ViralLoops_counterWidget"
              data-vl-widget="referralCountWidget"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data-test": "ViralLoops_counterWidget",
              "data-vl-widget": "referralCountWidget",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ViralLoops /> formWidget renders with a form section 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViralLoops
    formWidget={true}
    publicToken="HaOj3wBPfSJhoWkoW2Ad_rmvPoM"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED>
        <div
          data-test="ViralLoops_formWidget"
          data-vl-widget="embedForm"
        />
        
        
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            data-test="ViralLoops_formWidget"
            data-vl-widget="embedForm"
          />,
          "",
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data-test": "ViralLoops_formWidget",
            "data-vl-widget": "embedForm",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "",
        "",
      ],
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED>
          <div
            data-test="ViralLoops_formWidget"
            data-vl-widget="embedForm"
          />
          
          
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              data-test="ViralLoops_formWidget"
              data-vl-widget="embedForm"
            />,
            "",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data-test": "ViralLoops_formWidget",
              "data-vl-widget": "embedForm",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "",
          "",
        ],
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ViralLoops /> milestoneWidget renders with a rewards section 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViralLoops
    milestoneWidget={true}
    publicToken="HaOj3wBPfSJhoWkoW2Ad_rmvPoM"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED>
        
        <div
          data-test="ViralLoops_milestoneWidget"
          data-vl-widget="milestoneWidget"
        />
        
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "",
          <div
            data-test="ViralLoops_milestoneWidget"
            data-vl-widget="milestoneWidget"
          />,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data-test": "ViralLoops_milestoneWidget",
            "data-vl-widget": "milestoneWidget",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED>
          
          <div
            data-test="ViralLoops_milestoneWidget"
            data-vl-widget="milestoneWidget"
          />
          
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "",
            <div
              data-test="ViralLoops_milestoneWidget"
              data-vl-widget="milestoneWidget"
            />,
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data-test": "ViralLoops_milestoneWidget",
              "data-vl-widget": "milestoneWidget",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "",
        ],
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
